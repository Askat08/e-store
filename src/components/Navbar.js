import React from "react";
import { Link } from "react-router-dom";
import logo from "../logo.svg";

export default function Navbar() {
  return (
    <nav className="navbar navbar-expand-sm bg-dark   navbar-dark">
      <Link to="/">
        <img src={logo} alt="store-bag-logo" className="navbar-brand" />
      </Link>
      <ul className="navbar-nav align-items-center">
        <li className="nav-item ml-5">
          <Link to="/" className="nav-link text-info">
            <h4>comics</h4>
          </Link>
        </li>
      </ul>
      <Link to="/cart" className="ml-auto">
        <button className="btn btn-info">
          <i className="fas fa-cart-arrow-down" />
          {"             "}My Cart
        </button>
      </Link>
    </nav>
  );
}
