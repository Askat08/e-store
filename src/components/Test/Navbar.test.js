import React from "react";
import { shallow } from "enzyme";
import Navbar from "../Navbar";

describe("Navbar component test", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Navbar />);
  });

  it("should render Navbar component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test for Navbar logo", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toEqual("logo.svg");
  });

  it("should test cart icon inside button", () => {
    const button = wrapper.find("button");
    expect(button.contains(<i className="fas fa-cart-arrow-down" />)).toEqual(
      true
    );
  });

  it("includes link to Mission scene", () => {
    expect(wrapper.find("Link").at(0).props().to).toBe("/");
    expect(wrapper.find("Link").at(1).props().to).toBe("/");
    expect(wrapper.find("Link").at(2).props().to).toBe("/cart");
  });

  it("should test h4 text", () => {
    expect(wrapper.find("h4").text()).toEqual("comics");
  });
});
