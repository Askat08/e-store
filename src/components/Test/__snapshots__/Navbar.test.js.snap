// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component test should render Navbar component 1`] = `
<nav
  className="navbar navbar-expand-sm bg-dark   navbar-dark"
>
  <Link
    to="/"
  >
    <img
      alt="store-bag-logo"
      className="navbar-brand"
      src="logo.svg"
    />
  </Link>
  <ul
    className="navbar-nav align-items-center"
  >
    <li
      className="nav-item ml-5"
    >
      <Link
        className="nav-link text-info"
        to="/"
      >
        <h4>
          comics
        </h4>
      </Link>
    </li>
  </ul>
  <Link
    className="ml-auto"
    to="/cart"
  >
    <button
      className="btn btn-info"
    >
      <i
        className="fas fa-cart-arrow-down"
      />
                   
      My Cart
    </button>
  </Link>
</nav>
`;
